src/rpcrawtransaction.cpp

```diff
@@ -444,6 +445,14 @@ UniValue createrawtransaction(const UniValue& params, bool fHelp)
 
     CMutableTransaction rawTx = CreateNewContextualCMutableTransaction(
         Params().GetConsensus(), chainActive.Height() + 1);
+    
+    if (NetworkUpgradeActive(chainActive.Height(), Params().GetConsensus(), Consensus::UPGRADE_OVERWINTER)) {
+        if (chainActive.Height() + blockExpiry > TX_EXPIRY_HEIGHT_THRESHOLD){
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "nExpiryHeight must be less than TX_EXPIRY_HEIGHT_THRESHOLD.");
+        } else {
+            rawTx.nExpiryHeight = chainActive.Height() + blockExpiry;
```

Comment:
Duplication. Just set `rawTx.nExpiryHeight` to begin with, and then check it against the threshold. We immediately throw if it is invalid, so it won't leak anywhere.